x-pack/plugins/security/public/plugin.ts

```diff
@@ -28,18 +43,48 @@ export class SecurityPlugin implements Plugin<SecurityPluginSetup, SecurityPlugi
     this.sessionTimeout = new SessionTimeout(notifications, sessionExpired, http, tenant);
     http.intercept(new SessionTimeoutHttpInterceptor(this.sessionTimeout, anonymousPaths));
 
+    this.license$ = licensing.license$;
+
     return {
       anonymousPaths,
       sessionTimeout: this.sessionTimeout,
     };
   }
 
-  public start() {
+  public start(core: CoreStart) {
     this.sessionTimeout.start();
+
+    const securityLicenseService = new SecurityLicenseService().setup();
+    this.licenseSubscription = this.license$.subscribe(rawLicense => {
+      securityLicenseService.update(rawLicense);
+      const showSecurityLinks = securityLicenseService.license.getFeatures().showLinks;
+
+      const shouldRegisterNavControl =
+        !this.isAnonymousPath(core.http.anonymousPaths) && showSecurityLinks;
+
+      if (shouldRegisterNavControl && !this.navControlRegistered) {
+        const user = core.http.get('/api/security/v1/me', {
```

Comment:
Note - we can use the existing `UserAPI.getCurrentUser()` if we want to, once we migrate that from the LP.